lib/jekyll-titles-from-headings/generator.rb

```diff
@@ -5,12 +5,12 @@ class Generator < Jekyll::Generator
     attr_accessor :site
     TITLE_REGEX =
       %r!
-        \A\s*                   # Beginning and whitespace
-          (?:                   # either
-            \#{1,3}\s+(.*)      # atx-style header
-            |                   # or
-            (.*)\r?\n[-=]+\s*   # Setex-style header
-          )$                    # end of line
+        \A\s*                             # Beginning and whitespace
+          (?:                             # either
+            \#{1,3}\s+(.*)(?:\s+\#{1,3})? # atx-style header
```

Comment:
The number of hash marks in the optional suffix is actually arbitrary, so the should have been `(?:\s+\#+)?`.